### ISSUE TITLE
Release process makes release tag public while artefacts are still being validated

### ISSUE THREAD

**Issue by SamBarker** [kroxylicious/kroxylicious#1086]
**Describe the bug**
We create and publish the release tag as part of `stage-release.sh` which means it appears on GitHub while we are still validating that the release build is good. This is fine on the happy path. However it is problematic if we find issues with the build staged at maven central.

**To Reproduce**
1. run the stage-release action
2. decide the staged jars are problematic.
3. drop the staged repository.
4. The now stale release tag still exists in GitHub

**Expected behavior**

**YAML files and logs**

Attach or copy paste your kroxylicious configuration YAML (with sensitive details
like broker address redacted).
Attach or copy and paste relevant logs.

**Additional context**

**Comment by gracegrimwood** [comment]
We want to add tag cleanup to the GitHub Action and also start a discussion about release candidates.

**Comment by k-wall** [comment]
> The now stale release tag still exists in GitHub

I don't think this is true.  If you the `drop` a release, the release is deleted from GitHub.  This step removes the release notes and the tag.

**Comment by robobario** [comment]
@SamBarker we chatted about this in call, keen to hear if you used the deploy_release drop.

**Comment by SamBarker** [comment]
I think I'd tried deleting something by hand before running drop so that failed but the specifics have gone

**Comment by SamBarker** [comment]
> Publicly visible but marked as a draft.
The "release" is a draft but the git tag is fully public right? 

I think the basis of concern is are we creating the tag and thus the release at the wrong step? Should we actually tag as and create the release as part of the promotion step rather than as part of staging?

**Comment by k-wall** [comment]
> I think the basis of concern is are we creating the tag and thus the release at the wrong step? Should we actually tag as and create the release as part of the promotion step rather than as part of staging?

I suppose that would be possible.

We'd need to retrospectively tag the commit that was used to build the release as part of the build promotion phase.  To do that we'd need to work out the hash.  I don't think that would be too hard.

Also, the drafting step would need to be changed.  The https://github.com/kroxylicious/kroxylicious/releases/new workflow needs a tag, so drafting could no longer be part the the 'staging' step.   TBH I'd **love** to kill the manual changelog and switch to changelog generation (probably jreleaser:release).  Maintaining that CHANGELOG manually is mostly wasted effort (the merge conflicts are especially irksome).  We'd need to think about how we'd advertise deprecations.

**Comment by SamBarker** [comment]
While I agree with you @k-wall I think what we have is probably enough to close this issue. We can revisit change log generation etc later

**Comment by SamBarker** [comment]
@k-wall the latest release failed because the tag already existed: https://github.com/kroxylicious/kroxylicious/actions/runs/8610252056/job/23595478558#step:9:11 

I note the clean up is running `transition-staging-repository-state` rather than running `transition-github-release-note-state.sh` then running the `deploy_release` workflow with `drop` [also fails ](https://github.com/kroxylicious/kroxylicious/actions/runs/8610406949/job/23595871699) with the same error. 

That is drop doesn't work if there is no repo to drop. I think that should be fine and it should continue but I'm not quite sure whats the best way to implement that.

**Comment by k-wall** [comment]
Like i said above, I'd put energy into #1461  first. This should mean that we no longer need to tag so early in the release process.

**Comment by tombentley** [comment]
Note that because proxy image build is kicked off by the git tagging and because this happens before the `mvn deploy` is means that images will appear on quay.io before the maven artifacts are released (assuming I'm reading things right).

**Comment by k-wall** [comment]
> Note that because proxy image build is kicked off by the git tagging and because this happens before the `mvn deploy` is means that images will appear on quay.io before the maven artifacts are released (assuming I'm reading things right).

Yes, that's the current situation.

**Comment by robobario** [comment]
@k-wall reading the comments in #2463 it looks like this issue is still alive, are we going to tackle it for 0.14.0, or move to the next milestone?

**Comment by k-wall** [comment]
Bumped.

**Comment by robobario** [comment]
has hopped many milestones without being picked up, removing from milestone. and will remove `triaged` to discuss

**Comment by SamBarker** [comment]
While `Release process makes release tag public while artefacts are still being validated` is still valid this issue is stale and we need to revisit the release process in a wider sense.
